google/cloud/storage/blob.py

```diff
@@ -818,8 +818,8 @@ def _extract_headers_from_download(self, response):
                 method, digest = match.groups()
                 digests[method] = digest
 
-        self.crc32c = digests.get("crc32c", None)
-        self.md5_hash = digests.get("md5", None)
+        self.crc32c = digests.get("crc32c", self.crc32c)
+        self.md5_hash = digests.get("md5", self.md5_hash)
```

Comment:
Likely pedantic, but I would prefer to have the assignment to the local properties conditioned by the presence of the `X-Goog-Hash` header:  if it is present, but does not contain one of the elements, that element should be cleared as well.  So, I'm arguing for having lines 814-822 contained inside an `if x_goog_hash:` block.

Reply:
I like it. I think that better expresses the intent.

Change made in [this commit](https://github.com/googleapis/python-storage/pull/267/commits/f0e77dcab1caca2677b14cc91e72ffa266f8ad0a).